### PR TITLE
Ironic pod resource consumption report

### PR DESCRIPTION
**Pull request by namnx228**

### CODE REVIEW

File: ironic-pod-resource-limit/experiment-report.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	For CPU usage: rate(container_cpu_usage_seconds_total[10m]) * 1000
+
+`container_cpu_usage_seconds_total` measures the number of seconds one container uses CPU. In detail, if one container uses 1 second CPU in 1 second, that container uses 100% of one CPU core (1000 milicore) in that moment. Similarly, if it uses 0.5 second, it uses 500 milicore, and 2 seconds mean 2000 milicore. However, this metric is aggregating over time, so using it alone is not enough.
+
+`rate()[10m]` calculates the per-second average rate of increase of the time series of 10 minutes in the range vector. https://prometheus.io/docs/prometheus/latest/querying/functions/#rate. In short, combine with the above function, `rate()` measures the number of CPU seconds at any point in the timeline. The 10 minutes time range decides how sharp the plot is. The larger time range, the flatter the plot. 10-minute-duration is chosen to make the plot not too much frustrated, but also indicating some ideas about how much CPU a container can consume in a specific moment. To convert to milicore, a multiply to 1000 is needed. 
+
+
+
+## 4 BMHs 
+```bash
```
Comment (lentzi90) [inline ironic-pod-resource-limit/experiment-report.md:29]:
````suggestion
```
````

File: ironic-pod-resource-limit/experiment-report.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+![memory_4BMHs][5]
+Sum memory usage 
+![sum_memory_4BMHs][6]
+CPU usage by each container
+![cpu_4BMHs][7]
+Sum CPU usage
+![sum_cpu_4BMHs][8]
+
+## 10 BMHs 
+```bash
```
Comment (lentzi90) [inline ironic-pod-resource-limit/experiment-report.md:54]:
````suggestion
```
````

File: ironic-pod-resource-limit/experiment-report.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+Sum memory usage 
+![sum_memory_10BMHs][1]
+CPU usage by each container
+![cpu_10BMHs][3]
+Sum CPU usage
+![sum_cpu_10BMHs][4]
+
+
+## 15 BMHs 
+```bash
```
Comment (lentzi90) [inline ironic-pod-resource-limit/experiment-report.md:79]:
````suggestion
```
````

File: ironic-pod-resource-limit/experiment-report.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+Sum memory usage
+![sum_memory_15BMHs][10]
+CPU usage by each container
+![cpu_15BMHs][11]
+Sum CPU usage
+![sum_cpu_15BMHs][12]
+
+
+## 20 BMHs 
+```bash
```
Comment (lentzi90) [inline ironic-pod-resource-limit/experiment-report.md:104]:
````suggestion
```
````

File: ironic-pod-resource-limit/experiment-report.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+![memory_20BMHs][13]
+Sum memory usage
+![sum_memory_20BMHs][14]
+CPU usage by each container
+![cpu_20BMHs][15]
+Sum CPU usage
+![sum_cpu_20BMHs][16]
+
+## 30 BMHs 
+```bash
```
Comment (lentzi90) [inline ironic-pod-resource-limit/experiment-report.md:128]:
````suggestion
```
````

File: ironic-pod-resource-limit/experiment-report.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+Sum memory usage
+![sum_memory_30BMHs][18]
+CPU usage by each container
+![cpu_30BMHs][19]
+Sum CPU usage
+![sum_cpu_30BMHs][20]
+
+
+## 40 BMHs 
+```bash
```
Comment (lentzi90) [inline ironic-pod-resource-limit/experiment-report.md:153]:
````suggestion
```
````

**Review by lentzi90 — COMMENTED** [review]
Really nice looking!
What kind of tests are you running? Is it feature tests or just creating the cluster? This would be good to explain in the document also I think.

**Review by lentzi90 — APPROVED** [review]

File: ironic-pod-resource-limit/experiment-report.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+## Experiment environment
+Operating system: `Ubuntu 20.04.3 LTS`
+
+Management cluster: Minikube
+
+Metric collector: Prometheus in cluster
+
+Data visualizer: Grafana in cluster
+
+Merics:
```
Comment (furkatgofurov7) [inline ironic-pod-resource-limit/experiment-report.md:14]: *metrics

File: ironic-pod-resource-limit/experiment-report.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+	For CPU usage: rate(container_cpu_usage_seconds_total[10m]) * 1000
+
+`container_cpu_usage_seconds_total` measures the number of seconds one container uses CPU. In detail, if one container uses 1 second CPU in 1 second, that container uses 100% of one CPU core (1000 milicore) in that moment. Similarly, if it uses 0.5 second, it uses 500 milicore, and 2 seconds mean 2000 milicore. However, this metric is aggregating over time, so using it alone is not enough.
+
+`rate()[10m]` calculates the per-second average rate of increase of the time series of 10 minutes in the range vector. https://prometheus.io/docs/prometheus/latest/querying/functions/#rate. In short, combine with the above function, `rate()` measures the number of CPU seconds at any point in the timeline. The 10 minutes time range decides how sharp the plot is. The larger time range, the flatter the plot. 10-minute-duration is chosen to make the plot not too much frustrated, but also indicating some ideas about how much CPU a container can consume in a specific moment. To convert to milicore, a multiply to 1000 is needed.
```
Comment (furkatgofurov7) [inline ironic-pod-resource-limit/experiment-report.md:24]: *combined

File: ironic-pod-resource-limit/experiment-report.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+`rate()[10m]` calculates the per-second average rate of increase of the time series of 10 minutes in the range vector. https://prometheus.io/docs/prometheus/latest/querying/functions/#rate. In short, combine with the above function, `rate()` measures the number of CPU seconds at any point in the timeline. The 10 minutes time range decides how sharp the plot is. The larger time range, the flatter the plot. 10-minute-duration is chosen to make the plot not too much frustrated, but also indicating some ideas about how much CPU a container can consume in a specific moment. To convert to milicore, a multiply to 1000 is needed. 
+
+Experiment: This experiment is based on metal3 integration test. Prometheus is installed into Minikube cluster after it starts. It monitors all activities of Ironic pod in a integration test, including instropecting, provisioning,...
```
Comment (furkatgofurov7) [inline ironic-pod-resource-limit/experiment-report.md:26]: *introspecting

File: ironic-pod-resource-limit/experiment-report.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+```
+Memory usage by each container
+![memory_20BMHs][13]
+Sum memory usage
+![sum_memory_20BMHs][14]
+CPU usage by each container
+![cpu_20BMHs][15]
+Sum CPU usage
+![sum_cpu_20BMHs][16]
+
```
Comment (furkatgofurov7) [inline ironic-pod-resource-limit/experiment-report.md:126]: needs a new line

**Review by furkatgofurov7 — COMMENTED** [review]
Great write-up! Forgot to click on "Submit review" after reviewing this). Few nits inline:

**Pull request merged**